File: integrations/src/cinderscapes/java/juuxel/adorn/compat/cinderscapes/CinderscapesCompat.java
package juuxel.adorn.compat.cinderscapes;

import com.google.auto.service.AutoService;
import juuxel.adorn.block.variant.BlockVariant;
import juuxel.adorn.block.variant.CompatBlockVariantSet;
import juuxel.adorn.block.variant.PrefixedBlockVariantSet;

import java.util.List;

@AutoService(CompatBlockVariantSet.class)
public final class CinderscapesCompat extends PrefixedBlockVariantSet {
    @Override
    public String getModId() {
        return "cinderscapes";
    }

    @Override
    public List<BlockVariant> getWoodVariants() {
        return createVariants(
            BlockVariant.Wood::new,
            "scorched",
            "umbral"
        );
    }
}

File: src/main/kotlin/juuxel/adorn/compat/CinderscapesCompat.kt
package juuxel.adorn.compat

import juuxel.adorn.Adorn
import juuxel.adorn.api.block.AdornBlockBuilder
import juuxel.adorn.api.block.BlockVariant

object CinderscapesCompat {
    fun init() {
        val fungusTypes = sequenceOf(
            "scorched",
            "umbral",
        ).map { BlockVariant.Wood("cinderscapes/$it") }

        for (fungus in fungusTypes) {
            AdornBlockBuilder.create(fungus)
                .withEverything()
                .registerIn(Adorn.NAMESPACE)
        }
    }
}
